### PR TITLE
[MRG+1] Fix markup in plt.subplots docstring.

### LINKED ISSUE #8078: plt.subplots crashes when handed fig_kw argument
Current docs (http://matplotlib.org/devdocs/api/pyplot_api.html#matplotlib.pyplot.subplots) list `fig_kw` as one of the arguments that can be passed to `plt.subplots`, however this fails with:

```
import matplotlib.pyplot as plt
fig, axs = plt.subplots(1, 2, fig_kw={'figsize': (15, 8)})
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dstansby/matplotlib/lib/matplotlib/pyplot.py", line 1190, in subplots
    fig = figure(**fig_kw)
  File "/home/dstansby/matplotlib/lib/matplotlib/pyplot.py", line 544, in figure
    **kwargs)
  File "/home/dstansby/matplotlib/lib/matplotlib/backends/backend_qt5agg.py", line 43, in new_figure_manager
    thisFig = FigureClass(*args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'fig_kw'
```

~~Either~~ the docs should be updated to note that anything extra handed to `plt.subplots` is handed to `plt.figure`~~, or `fig_kw` should be an allowed argument.~~

### PR DESCRIPTION
**Pull request by anntzer**
(does not apply to Figure.subplots, which does not take fig_kw).

closes #8078 
see http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html for markup (numpydoc itself does not have an example for that but I hope that it works too...)

### CODE REVIEW

**Comment by QuLogic** [comment]
We do it with `**kwargs`, so it should work; maybe also update the description to remove the `dict` type, which isn't really true?

**Comment by anntzer** [comment]
sounds reasonable, fixed

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -1125,9 +1125,9 @@ def subplots(nrows=1, ncols=1, sharex=False, sharey=False, squeeze=True,
         :class:`~matplotlib.gridspec.GridSpec` constructor used to create the
         grid the subplots are placed on.
 
-    fig_kw : dict, optional
-        Dict with keywords passed to the :func:`figure` call.  Note that all
-        keywords not recognized above will be automatically included here.
+    **fig_kw :
+        All additional keywords arguments are passed to the :func:`figure`
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: keyword

Comment (anntzer) [inline lib/matplotlib/pyplot.py]: yup

**Pull request merged**